pytorch_lightning/trainer/callback_hook.py

```diff
@@ -302,6 +303,11 @@ def on_load_checkpoint(self, checkpoint):
                 UserWarning,
             )
 
+        # in case QAT was used for original training and not inserted void  after resume
+        is_qat = any(isinstance(cb, QuantizationAwareTraining) for cb in self.callbacks)
+        if QuantizationAwareTraining in callback_states and not is_qat:
+            self.callbacks.append(QuantizationAwareTraining())
+
```

Comment:
i am very opposed to special casing callbacks like this

Reply:
Would you prefer to raise a MisConfigurationError instead ? From my understanding, this is the only other alternative.

Reply:
ok, I'll change it as misconfiguration :rabbit:

Reply:
@Borda  Mind raising a MisConfigurationError ?